js/web/lib/wasm/jsep/webgpu/ops/common.ts

```diff
@@ -194,9 +194,8 @@ export interface IndicesHelper {
 const getWgslMappedType = (type: number, components: 1|2|3|4): string|[string, string] => {
   // return type is [ storage type, runtime type ] or a single string for both
   switch (type) {
-    // TODO: enable after "shader-f16" WSGL extension release
-    // case DataType.float16:
-    //   return components > 1 ? `vec${components}<f16>` : 'f16';
+    case DataType.float16:
+      return components > 1 ? `vec${components}<f16>` : 'f16';
```

Comment:
did you mention that `vec3<f16>` is invalid type (because of alignment or something)? if so, we can throw an error so that it's easier to figure out.

Reply:
yeah. vec3<f32> has alignment of 16 bytes so basically it is a vec4 with last element set to 0. same is with f16 or any other type, so it cannot be used as input/output array. will add a check